xarray/tests/test_backends.py

```diff
@@ -552,6 +552,19 @@ def test_ondisk_after_print(self):
             repr(on_disk)
             assert not on_disk['var1']._in_memory
 
+    def test_setncattr_string(self):
```

Comment:
You might need to add this test specifically for the class with netCDF4 tests, making use of the `self.roundtrip()` method. Otherwise this is also going to run for cases where the list of strings attribute cannot be roundtripped (e.g., writing a netCDF3 file).